### PR TITLE
Address mock proofing calls (LG-3275)

### PR DESCRIPTION
**Pull request by mitchellhenke**
Sibling PR to https://github.com/18F/identity-idp/pull/4273/files (which will use the results of this PR)

Adds mock address proofing class and a lambda to use that, then the IDP can decide which vendor it wants to use through the lambda

I created a new structure for mock proofers because we'll have a couple more of those soon

### CODE REVIEW

File: source/proof_address_mock/lib/proof_address_mock.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+require 'faraday'
+require 'retries'
+require 'identity-idp-functions/mock_proofers/address_mock'
+
+module IdentityIdpFunctions
+  class ProofAddressMock
+    def self.handle(event:, context:, &callback_block)
+      params = JSON.parse(event['body'], symbolize_names: true)
+      new(
+        idp_api_auth_token: ENV['IDP_API_AUTH_TOKEN'],
```
Comment (zachmargolis) [inline source/proof_address_mock/lib/proof_address_mock.rb:11]: let's make this `ENV.fetch` so we error if it's missing a value

**Review by zachmargolis — COMMENTED** [review]
We need to update `source/template.yaml` and essentially copy-paste `DemoFunction` (and `Demo Function` and `demo_function` for this new one.

Since lambdas only have one entry point if we want to expose all the mock proofers, I think we need to make a separate lambda for each of them (so just `IdentityIdpFunctions::AddressMock` or even `IdentityIdpFunctoins::MockProofAddress` or something

File: source/proof_address_mock/lib/proof_address_mock.rb (hunk 1 above)
Comment (mitchellhenke) [reply source/proof_address_mock/lib/proof_address_mock.rb:11]: I made it optional because it doesn't get set in development/test environments.

We could set it in those environments, but we'd need a different name for the keys since there are different auth tokens for the address and resolution API endpoints.

Comment (zachmargolis) [reply source/proof_address_mock/lib/proof_address_mock.rb:11]: Oh good point!

**Comment by mitchellhenke** [comment]
@zachmargolis updated `template.yaml` with the address mock in 6d681f92cf2a475758fdc5ae8fe584e71737c083, but not sure if it's right

**Comment by mitchellhenke** [comment]
Hm, getting some build errors that are hard to parse: https://app.circleci.com/pipelines/github/18F/identity-idp-functions/27/workflows/33a5345a-aad2-4259-950a-91c89175b1fd/jobs/25

File: lib/identity-idp-functions/mock_proofers/address_mock.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+require 'proofer'
+
+module IdentityIdpFunctions
+  module MockProofers
+    class AddressMock < Proofer::Base
+      vendor_name 'AddressMock'
```
Comment (zachmargolis) [inline lib/identity-idp-functions/mock_proofers/address_mock.rb:6]: Ok so I *think* this may cause issues --- every directory under `source/` is deployed as its own folder (hence their own Gemfile, etc) so having this logic get pulled in from outside the `source/` may not work ... I think we need to move this inside.

I *think* the `aws-ruby-sdk` folder is an example of sharing code across multiple lambdas, so if we did want to share this in multiple lambdas, I think we'd have to do something like that and update the `template.yaml` to include things appropriately

**Comment by zachmargolis** [comment]
> Hm, getting some build errors that are hard to parse: https://app.circleci.com/pipelines/github/18F/identity-idp-functions/27/workflows/33a5345a-aad2-4259-950a-91c89175b1fd/jobs/25

Looks like an issue with installing a dependency via homebrew..... @crissupb have you seen this before? Is it transient?

File: lib/identity-idp-functions/proof_address_mock.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+require 'identity-idp-functions'
+
+# This is a trampoline to allow lazily loading the proof_address function
```
Comment (zachmargolis) [inline lib/identity-idp-functions/proof_address_mock.rb:3]: comment is a lie, should be: `proof_address_mock`

File: source/proof_address/lib/proof_address.rb [hunk 4]
```diff
@@ -26,8 +26,12 @@ def proof(&callback_block)
         lexisnexis_proofer.proof(applicant_pii)
       end
 
+      result = proofer_result.to_h
+      result[:context] = { stages: [address: LexisNexis::PhoneFinder::Proofer.vendor_name] }
```
Comment (zachmargolis) [inline source/proof_address/lib/proof_address.rb:30]: thanks for adding these to fix

**Review by zachmargolis — APPROVED** [review]
LGTM! Let's bump `version.rb` while we're here too

**Comment by mitchellhenke** [comment]
The CI failure is still happening, but I was curious if it was one of the changes I made or something else.

I reran the CI on the master branch [here](https://app.circleci.com/pipelines/github/18F/identity-idp-functions/23/workflows/972ccf27-ae83-4e38-a11f-58b2c8f00655/jobs/30) and it fails in the same way, so something else is up

File: .circleci/config.yml [hunk 5]
```diff
@@ -30,9 +30,10 @@ commands:
           command: |
             PATH=/home/linuxbrew/.linuxbrew/bin:$PATH
             brew tap aws/tap
-            brew install aws-sam-cli
+            # brew install aws-sam-cli
```
Comment (aduth) [inline .circleci/config.yml:33]: Do we want to leave this commented, or just remove it?

```suggestion
```
